core/src/main/java/org/apache/accumulo/core/iterators/OrIterator.java

```diff
@@ -249,4 +268,9 @@ final public boolean hasTop() {
   public void init(SortedKeyValueIterator<Key,Value> source, Map<String,String> options, IteratorEnvironment env) throws IOException {
     throw new UnsupportedOperationException();
   }
+
+  @Override
+  public void close() throws Exception {
+    sources.forEach(s -> s.closeSafely());
+  }
```

Comment:
For exception handling of multiple sources, you thinking of not using forEach and going old school loop?